whisper/whisperv5/whisper.go

```diff
@@ -617,6 +626,13 @@ func (wh *Whisper) add(envelope *Envelope) (bool, error) {
 		}
 	}
 	wh.poolMu.Unlock()
+	wh.peerMu.RLock()
+	if !alreadyCached {
```

Comment:
It would be better to put the lock and unlock inside the `if` so that there is no unnecessary locking.